index.js

```diff
@@ -7,41 +7,17 @@ import { getData } from './api/requests.js';
 
 const tableContainer = document.getElementById('table');
 
-const countries = await getData();
+const firstPage = await getData(20, 1);
+const totalAmount = await getData();
 
-const countriesLenght = Object.keys(countries).length;
-const totalAmount = countriesLenght;
-
-const callbacksObject = {
-  sortCallback: (dataKey, isAsc) => {
-    const isDataKey = Boolean(dataKey);
-    if (!isDataKey) {
-      return countriesTable.render(countries, totalAmount);
-    }
-    isAsc
-      ? countriesTable.render(quickSort(countries, dataKey), totalAmount)
-      : countriesTable.render(quickSort(countries, dataKey).reverse(), totalAmount);
-  },
-  paginationCallback: (currentPage, amountEl) => {
-    if (!currentPage && !amountEl) {
-      return countriesTable.render(countries, totalAmount);
-    }
-    let firstElOnPage = (currentPage - 1) * amountEl;
-    let lastElOnPage = firstElOnPage + amountEl;
-    let paginationData = countries.slice(firstElOnPage, lastElOnPage);
-
-    countriesTable.render(paginationData, totalAmount);
-  },
-  filterCallback: (filter, dataKey) => {
-    if (!filter && !dataKey) {
-      return countriesTable.render(countries, totalAmount);
-    }
-    const sortedArr = countries.filter((country) => country[dataKey].toLowerCase().indexOf(filter.toLowerCase()) > -1);
-    countriesTable.render(sortedArr, sortedArr.length);
-  },
+const getDataCallback = async (amountEl, currentPage, isAsc, dataKey, filter) => {
+  const ascFlag = isAsc ? 'asc' : 'desc';
+  const sortRequest = await getData(amountEl, currentPage, ascFlag, dataKey, filter);
```

Comment:
Please, use some more common variable names. This could be not sorting requests only